tests/integration/backends/test_tpp.py

```diff
@@ -1484,6 +1485,123 @@ def test_registered_tests_are_exhaustive():
+                CodedEvent(Patient_ID=patient_id, CTV3Code=value)
+            ],
+        ),
+        (
+            tpp.clinical_events,
+            tpp.clinical_events.snomedct_code,
+            ["123000", "123001", "123002", "123003"],
+            lambda patient_id, value: [
+                CodedEvent_SNOMED(Patient_ID=patient_id, ConceptId=value)
+            ],
+        ),
+        (
+            tpp.medications,
+            tpp.medications.dmd_code,
+            ["123000", "123001", "123002", "123003"],
+            lambda patient_id, value: [
+                MedicationDictionary(MultilexDrug_ID=f";{value};", DMD_ID=value),
+                MedicationIssue(Patient_ID=patient_id, MultilexDrug_ID=f";{value};"),
+            ],
+        ),
+        (
+            tpp.open_prompt,
+            tpp.open_prompt.ctv3_code,
+            ["abc00", "abc01", "abc02", "abc03"],
+            lambda patient_id, value: [
+                OpenPROMPT(Patient_ID=patient_id, CTV3Code=value)
+            ],
+        ),
+        (
+            tpp.open_prompt,
+            tpp.open_prompt.snomedct_code,
+            ["123000", "123001", "123002", "123003"],
+            lambda patient_id, value: [
+                OpenPROMPT(Patient_ID=patient_id, ConceptId=value, CodeSystemId=0)
+            ],
+        ),
+    ],
+)
+def test_is_in_queries_on_columns_with_nonstandard_collation(
+    mssql_engine, table, column, values, factory
+):
+    # Assign a patient ID to each value
+    patient_values = list(enumerate(values, start=1))
+    # Create patient data for each of the values
+    mssql_engine.setup(
+        [factory(patient_id, value) for patient_id, value in patient_values]
+    )
+    # Choose every other value to match against (so we have a mixture of matching and
+    # non-matching patients)
+    matching_values = values[::2]
+
+    dataset = create_dataset()
+    dataset.define_population(table.exists_for_patient())
+    dataset.matches = table.where(column.is_in(matching_values)).exists_for_patient()
+    results = mssql_engine.extract(
+        dataset,
+        # Configure query engine to always break out lists into temporary tables so we
+        # exercise that code path
+        config={"EHRQL_MAX_MULTIVALUE_PARAM_LENGTH": 1},
+        backend=TPPBackend(
+            config={"TEMP_DATABASE_NAME": "temp_tables"},
+        ),
+    )
+
+    # Check that the expected patients match
+    assert results == [
+        {"patient_id": patient_id, "matches": value in matching_values}
+        for patient_id, value in patient_values
+    ]
+
+
+def test_nonstandard_collation_test_is_exhaustive():
+    # Walk over every ORM class we use in this test module and collect all the columns
+    # on those classes which use a nonstandard collation. We don't inspect every class
+    # in the schema because that contains tables we don't currently use. But the moment
+    # we start using a table we want it to be included.
```

Comment:
This is a bit weird, and only works (I think?) because we have `test_registered_tests_are_exhaustive`.  Could you add a comment explaining this?

Reply:
Yes, that's right (and yes it is more than a bit weird). I'll try to clarify in the comments.

The alternative approach I thought of was to walk the full schema and have an explicit list of columns we don't use. But my worry was that nothing keeps that up-to-date if we start using one of those columns. Piggy backing off the imports (in combination with `test_registered_tests_are_exhaustive`) was the only way I could come with of capturing this automatically.